pkg/cloudprovider/aws/launchtemplate.go

```diff
@@ -211,6 +218,20 @@ func (p *LaunchTemplateProvider) createLaunchTemplate(ctx context.Context, optio
 	return output.LaunchTemplate, nil
 }
 
+func (p *LaunchTemplateProvider) onCacheEvicted(key string, lt interface{}) {
```

Comment:
I guess we leak if we miss evictions?

Reply:
Yes, and this can happen when Karpenter shuts down...  I thought about adding a shutdown hook and draining the cache... but I'm not sure it's worth it to do that.  

Another option I just thought of is loading all LTs into the cache that Karpenter has created by querying the Karpenter tag so that when Karpenter starts back up, it will load in all LTs and then they will naturally be evicted if they are not used.

Reply:
I quite like this, actually. Hydrate the LT cache on startup. We can panic if it fails, since it's initialization logic.

Reply:
👍  Cool, I'll do the cache hydration. Since we don't currently tag the LT with anything specific, this will only work for LTs created after this update is installed :(